src/platform_impl/macos/view.rs

```diff
@@ -431,62 +423,32 @@ extern "C" fn set_marked_text(
 ) {
     trace!("Triggered `setMarkedText`");
     unsafe {
-        let state_ptr: *mut c_void = *this.get_ivar("winitState");
-        let state = &mut *(state_ptr as *mut ViewState);
-
-        // Delete previous marked text
-        let char_count = ns_string_char_count(state.marked_text.string());
-        delete_marked_text(state, char_count);
-
-        state.is_ime_activated = true;
-
-        let _: () = msg_send![state.marked_text, release];
-        state.marked_text = NSMutableAttributedString::alloc(nil);
+        let marked_text_ref: &mut id = this.get_mut_ivar("markedText");
+        let _: () = msg_send![(*marked_text_ref), release];
+        let marked_text = NSMutableAttributedString::alloc(nil);
         let has_attr = msg_send![string, isKindOfClass: class!(NSAttributedString)];
         if has_attr {
-            state.marked_text.initWithAttributedString(string);
+            marked_text.initWithAttributedString(string);
         } else {
-            state.marked_text.initWithString(string);
+            marked_text.initWithString(string);
         };
-
-        let text_ns_str = state.marked_text.string();
-        let slice = slice::from_raw_parts(
-            text_ns_str.UTF8String() as *const c_uchar,
-            text_ns_str.len(),
-        );
-        let text_str = str::from_utf8_unchecked(slice);
-
-        for character in text_str.chars() {
-            AppState::queue_event(EventWrapper::StaticEvent(Event::WindowEvent {
-                window_id: WindowId(get_window_id(state.ns_window)),
-                event: WindowEvent::ReceivedCharacter(character),
-            }));
-        }
+        *marked_text_ref = marked_text;
     }
     trace!("Completed `setMarkedText`");
 }
 
-extern "C" fn unmark_text(this: &mut Object, _sel: Sel) {
+extern "C" fn unmark_text(this: &Object, _sel: Sel) {
     trace!("Triggered `unmarkText`");
     unsafe {
-        clear_marked_text(this);
+        let marked_text: id = *this.get_ivar("markedText");
+        let mutable_string = marked_text.mutableString();
+        let _: () = msg_send![mutable_string, setString:""];
```

Comment:
Not sure whether this or the other method (release + alloc) is better here, but let's just leave it be for now

Reply:
I agree that releasing and allocating a new one would be safer, but the reference to `markedText` seems to be constrainted to the View, so it seems safe enough.